### PR TITLE
test.migration: add test for indexeddb adapter

### PR DESCRIPTION
**Pull request by alxndrsn**
Test version upgrades with the `indexeddb` adapter.

Might be helpful to ignore whitespace when reviewing: https://github.com/pouchdb/pouchdb/pull/8752/files/?w=1

### CODE REVIEW

File: tests/integration/test.attachments.js [hunk 1]
```diff
@@ -3113,15 +3113,9 @@ adapters.forEach(function (adapter) {
...
-            document.body.appendChild(script);
-            var timeout = setInterval(function () {
-              if (window.bigimage) {
-                clearInterval(timeout);
-                cb(null, window.bigimage);
-              }
-            }, 500);
+            testUtils.asyncLoadScript('deps/bigimage.js')
+                .then(() => cb(null, window.bigimage))
+                .catch(err => cb(err));
```
Comment (alxndrsn) [inline tests/integration/test.attachments.js:3118]: Already submitted as separate PR.

File: tests/common-utils.js [hunk 2]
```diff
@@ -140,6 +140,7 @@ commonUtils.asyncLoadScript = function (url) {
         script.onload();
       }
     };
+    script.onerror = reject;
```
Comment (alxndrsn) [inline tests/common-utils.js:143]: Not directly relevant to this PR.  This needs testing separately, and including as its own PR if it works.

File: packages/node_modules/pouchdb-adapter-indexeddb/src/setup.js [hunk 3]
```diff
@@ -127,6 +132,32 @@ function openDatabase(openDatabases, api, opts, resolve, reject) {
...
+        doc.data.migrated = true;
+        Object.values(doc.revs).forEach(rev => {
+          rev.data.migrated = true;
+        });
+
+        cursor.update(doc).onsuccess = () => {
+          cursor.continue();
+        };
+      };
+    }
```
Comment (alxndrsn) [inline packages/node_modules/pouchdb-adapter-indexeddb/src/setup.js:160]: This all needs to be rolled back.

File: tests/integration/browser.migration.js [hunk 4]
```diff
@@ -1,27 +1,34 @@
...
+  const scenarios = [
+    { scenario: 'PouchDB v1.1.0', constructorName: 'PouchDBVersion110'} ,
+    { scenario: 'PouchDB v2.0.0', constructorName: 'PouchDBVersion200'} ,
+    { scenario: 'PouchDB v2.2.0', constructorName: 'PouchDBVersion220'} ,
+    { scenario: 'PouchDB v3.0.6', constructorName: 'PouchDBVersion306'} ,
+    { scenario: 'PouchDB v3.2.0', constructorName: 'PouchDBVersion320'} ,
+    { scenario: 'PouchDB v3.6.0', constructorName: 'PouchDBVersion360'} ,
+    { scenario: 'PouchDB v7.3.1', constructorName: 'PouchDBVersion731'} ,
+    { scenario: 'PouchDB v8.0.1', constructorName: 'PouchDBVersion801'} ,
+    { scenario: 'websql',         constructorName: 'PouchDB'} ,
```
Comment (alxndrsn) [inline tests/integration/browser.migration.js:31]: Already submitted in separate PR.

File: tests/integration/browser.migration.js [hunk 5]
```diff
@@ -631,523 +741,508 @@ describe('migration', function () {
...
+              .then(()    => oldPouch.get('norm-1'))
+              .then((doc) => oldPouch.putAttachment('norm-1',       'att.txt', doc._rev,  attFor('norm-1'), 'text/plain'))
+
+              .then(()    => oldPouch.putAttachment('norm-2',       'att.txt', undefined, attFor('norm-2'), 'text/plain'))
+
+              .then(() => oldPouch.close())
+
+              .then(() => {
+                newPouch = new dbs.second.pouch(dbs.second.local, { auto_compaction: false, adapter: 'indexeddb' });
+              })
```
Comment (garethbowen) [inline tests/integration/browser.migration.js:1197]: I can't help but think this sort of setup should be common to all migration tests but this implementation looks quite different from the "idb" one. Ideally as it's an integration test it would be more black box style testing where the "given" and "then" is the same, it's just the parameters for the "when" (adapter, pouch version, etc) that change. Is there a clean way to pull this out so we're testing the same things between adapters?

**Review by garethbowen — CHANGES_REQUESTED** [review]
Nice one! One suggestion inline but it may be too much work for this PR.

File: tests/integration/browser.migration.js (hunk 5 above)
Comment (alxndrsn) [reply tests/integration/browser.migration.js:1197]: > Is there a clean way to pull this out so we're testing the same things between adapters?

How about https://github.com/pouchdb/pouchdb/pull/8766?

**Comment by alxndrsn** [comment]
Now #8766 has been merged, this will need re-working to just introduce a new test around local docs & attachments.

**Comment by alxndrsn** [comment]
Superseded by https://github.com/pouchdb/pouchdb/pull/8766
